src/mlpack/core/optimizers/nestrov/nestrov_update.hpp

```diff
@@ -0,0 +1,107 @@
+/**
+ * @file nestrov_update.hpp
+ * @author Kris Singh
+ *
+ * Nestrov Momentum update for Stochastic Gradient Descent.
+ * 
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_SGD_NESTROV_UPDATE_HPP
+#define MLPACK_CORE_OPTIMIZERS_SGD_NESTROV_UPDATE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * Construct the NAG optimizer with the given function and parameters.  The
+ * defaults here are not necessarily good for the given problem, so it is
+ * suggested that the values used be tailored to the task at hand.  The
+ * maximum number of iterations refers to the maximum number of points that
+ * are processed (i.e., one iteration equals one point; one iteration does not
+ * equal one pass over the dataset). Typically the momentum paramter is often
+ * initialized with small value like 0.5 and later raised.
+
+ * @misc{1212.0901,
+ * Author = {Yoshua Bengio and Nicolas Boulanger-Lewandowski and Razvan Pascanu},
+ * Title = {Advances in Optimizing Recurrent Networks},
+ * Year = {2012},
+ * Eprint = {arXiv:1212.0901},
+ * } 
+ *
+ * @code
+ * @book{Goodfellow-et-al-2016,
+ *  title={Deep Learning},
+ *  author={Ian Goodfellow and Yoshua Bengio and Aaron Courville},
+ *  publisher={MIT Press},
+ *  note={\url{http://www.deeplearningbook.org}},
+ *  year={2016}
+ * }
+ */
+class NestrovUpdate
+{
+ public:
+  /**
+   * Construct the momentum update policy with given momentum decay parameter.
+   *
+   * @param momentum The momentum decay hyperparameter
+   */
+  NestrovUpdate(const double momentum = 0.5) : momentum(momentum)
+  { /* Do nothing. */ };
+
+  /**
+   * The Initialize method is called by SGD Optimizer method before the start of
+   * the iteration update process.  In the momentum update policy the velocity
+   * matrix is initialized to the zeros matrix with the same size as the
+   * gradient matrix (see mlpack::optimization::SGD::Optimizer )
+   *
+   * @param n_rows number of rows in the gradient matrix.
```

Comment:
Please use ``rows`` and ``cols`` instead of ``n_rows`` and ``n_cols`` here to match the rest of the codebase.